### ISSUE TITLE
EasyEffects crashes when microphone is muted

### ISSUE THREAD

**Issue by zastrixarundell** [wwmm/easyeffects#2828]
### EasyEffects Version

7.1.3

### What package are you using?

Arch (easyeffects)

### Distribution

Fedora Kinoite 39, uBlue

### Describe the bug

When the microphone is muted via the mute button on KDE (I have not tried on Gnome) EasyEffects would crash. This started happening fairly recently, maybe 2 weeks ago.

https://github.com/wwmm/easyeffects/assets/30195425/7d013022-3792-4fa5-baa8-68057c7442fb

### Expected Behavior

It won't crash.

### Debug Log

[Debug Log](https://github.com/wwmm/easyeffects/files/13769315/log.txt) (the file is too large to just paste).

**Comment by zastrixarundell** [comment]
So the minimum setup to crash when muting the microphone at least in my experiments are (all default configs):

* Noise Reduction
* Crystalizer
* Bass enhances
* Crystalizer

![image](https://github.com/wwmm/easyeffects/assets/30195425/7c164b48-dbfe-46f9-9df4-c3d6b48d31e1)

**Comment by zastrixarundell** [comment]
This might be #2752 potentially?

**Comment by wwmm** [comment]
>This might be https://github.com/wwmm/easyeffects/issues/2752 potentially?

Maybe. But as you've noticed the error messages are not exactly the same. But it is worth to install the development package available at the bottom of the following page https://github.com/wwmm/easyeffects/actions/runs/7322809280. I have already updated the `crystalizer` and the `bass enhancer` plugins with some changes that should fix #2752. So if your problem is the same it may help.

**Comment by wwmm** [comment]
But it is really strange that a mute mic causes this. Usually the result of this action is the processing of silence. So considering the crash is related to an avx call like `__memmove_avx_unaligned_erms` maybe one of the libraries EasyEffects is using did not like the values that came as a result of silence. Is there any change if you remove the crystalizer plugin from the pipeline? Or maybe the noise reduction plugin.

**Comment by zastrixarundell** [comment]
I just installed that version of EE and it isn't working, it's not processing my input audio at all. I get a blink of connected nodes on [qpwgraph](https://github.com/rncbc/qpwgraph) for maybe a second (no audio) and then everything disconnects. One time I *did* get some *some* audio but it was just a robotic sound on repeat, like a high voltage transforming station which is under HEAVY noise-altering plugins.

I did mess around with configuring EE and it seems that it crashes if the number of plugins overall is `>= 4` and both `Noise Reduction` and `Crystalizer` are present. 

So this would crash:
![image](https://github.com/wwmm/easyeffects/assets/30195425/19ea4629-9a4b-4943-bea1-9b07505f2c62)

I don't even know what LM does and I've never used it.

This and the following won't crash despite having multiple plugins:
![+crys-noise](https://github.com/wwmm/easyeffects/assets/30195425/2c7dc22b-9b9b-41f5-86d0-d4a85b7598d6)
![-crys+noise](https://github.com/wwmm/easyeffects/assets/30195425/207987cc-5b1a-4549-a648-1928d43e924d)

**Comment by wwmm** [comment]
>I don't even know what LM does and I've never used it.

It is not a plugin intended to be always in the pipeline. It is useful for people that are tuning the volume levels in some situations. Bu once it is in the desired point the plugin is removed. It does not change the audio data. It just estimate loudness levels.

**Comment by wwmm** [comment]
>This and the following won't crash despite having multiple plugins

Strange. Did you notice if the crash only happens when high sampling rates like the `96 kHz` in the image are used?

**Comment by zastrixarundell** [comment]
Okay seems that lowering it down to 48KHz did fix the crash although the tone of my voice when recording is changed with the same plugins, just by lowering it down to 48KHz.

I did have my system run @ 96KHz before this started crashing though so it's not the direct cause probably.

**Comment by wwmm** [comment]
>I did have my system run @ 96KHz before this started crashing though so it's not the direct cause probably

Strange indeed. And I do not remember a reason for 96 kHz to be a problem. The noise reduction plugin library only works with 48 kHz. So it will have to do resampling when the pipeline is working with a different rate. I wonder if the resampling process is somehow introducing bad data in the audio buffers that could make `__memmove_avx_unaligned_erms`. But as it isn't a function directly called by EasyEffects code I am not even sure about what is calling it exactly...

**Comment by zastrixarundell** [comment]
Probably better to link this issue to upstream at the noise library. But weird that it crashes where there isn't any audio source present as opposed for when there is.

**Comment by wwmm** [comment]
>But weird that it crashes where there isn't any audio source present as opposed for when there is.

I agree. It is not making any sense to me. Specially because I use the same plugins daily on my computer without crashes. At this moment I am at a videconference on MS Teams using both the speakers and the mic pipeline without problems. An with plenty of silent minutes. So the silence alone is not the problem. Something else is going on together with it.

**Comment by wwmm** [comment]
So much has changed that is probably better to close this issue
